feat: 新增Trello_api自動化文件

GeorgeMul opened the pull request:
--新增創建看板api
--建立logger
--建立api自動化文件夾
--setUp 取得id
--tearDown 刪除看板

Commit:
feat: 新增Trello_api自動化文件

--新增創建看板api
--建立logger
--建立api自動化文件夾
--setUp 取得id
--tearDown 刪除看板

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -134,10 +136,25 @@ def __init__(self, api_root_url):
 
     # post方法創建看板並且在api_root_url引入domain
-    def create_board_post(self, params):
+    def create_board(self, params):
```
建議命名http method放前面
post_create_board

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -134,10 +136,25 @@ def __init__(self, api_root_url):
         return res
 
+    # post方法創建自訂欄位在看板上
+    def create_Custom_Field(self, params, payload):
```
建議命名http method放前面

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -17,6 +17,8 @@
 # env.ini
 trello_URL = "https://api.trello.com/1"
 trello_Board_name = "My_test_board"
+key = ""
```
為啥這邊是空的

GeorgeMul replied:
key是空的原因是github機制,如果我放key會不讓我上版,我會改成 "請輸入您的key"

VincentSingGo replied:
喔喔咦所以你這個沒辦法拉下來就直接執行嗎

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -149,12 +166,49 @@ def __init__(self, config_url):
+            "key": key,
+            "token": token
+        }
+        body = {
```
不要寫body要馬寫payload要馬寫json

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -149,12 +166,49 @@ def __init__(self, config_url):
+        body = {
+            "idModel": board_id,
+            "modelType": "board",
+            "name": "Priority",
```
name要寫死嗎？

GeorgeMul replied:
好的,我再更改

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
-        # 驗證狀態為200
-        response = self.api_operation.create_trello_board()
-        assert response.status_code == 200
+    def create_board(self):
```
你是要create_board還是拿id?

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
+        return Custom_Field_ID
+
+    # 取得看板
+    def get_trello_board(self):
```
為啥每個method都要return id

VincentSingGo replied:
這個還是沒有告訴我

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
-        response_json = response.json()
-        return response_json
+        Custom_Field_ID = self.api_operation.create_trello_Custom_Field_on_board(Board_ID).json().get("id")
+        return Custom_Field_ID
```
這是變數命名方式應該是custom_field_id

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
+    # 刪除看板
+    def delete_trello_board(self, Board_ID):
+        response = self.api_operation.delete_trello_board(Board_ID)
+        assert response.status_code == 200
```
assert另外寫方法，感覺assert 200沒有什麼用主要是你要驗證res payload裡面的東西

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
+        return response
+
+    # 創建自訂欄位
+    def create_trello_Custom_Field_on_board(self):
```
為啥每個都要return board_id

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
+    def create_trello_Custom_Field_on_board(self):
+        board_id = self.create_board()
+        self.create_trello_Custom_Field(board_id)
+        return board_id
```
為啥每個都要return board_id

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
 # ----------------------------------------------------------------
+# confest 進行case操作前設定,引入url
+@pytest.fixture(autouse=True)
+def setUp_and_tearDown():
```
你這命名如果是針對單一case不能這樣命名

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
+    trello_domain = trello_URL
+    yield trello_domain
+    api_service = ApiService(trello_domain)
+    board_id = api_service.get_trello_board()
```
不要在setup teardown做邏輯，在service裡面包好再來用

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
 
-# case業務邏輯層
 class TestTrelloAPI:
+    @pytest.fixture(autouse=True)
```
我不建議每個都autouse你這樣case多會掛掉

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
 class TestTrelloAPI:
+    @pytest.fixture(autouse=True)
+    def init_fixtures(self, setUp_and_tearDown):
+        self.api_service = ApiService(setUp_and_tearDown)
```
可以在setup new

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -164,25 +218,53 @@ def __init__(self, domain):
-        api_service = ApiService(trello_domain)
-        # 執行case
-        api_service.create_board_use_post()
+        self.api_service.create_trello_Custom_Field_on_board()
```
你不寫case註解是要讓我猜你case在幹嘛嗎，還有你只有create_trello_Custom_Field_on_board你又是驗證了什麼

Commit:
fix: 更改程式碼邏輯

-- 更改命名名稱方法放前面
-- 更改autouse改成scope"module"
-- service從setUp進new出來
-- 更改case註解命名方式
-- 刪除assert 200 (後續研究後進行追加)
-- 更改自訂欄位命名不寫死
-- 更改內容body改成paylode
-- 刪除多餘回傳看板id
-- ini裡的key/token改成自行輸入內容

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -136,12 +137,12 @@ def __init__(self, api_root_url):
 
     # post方法創建自訂欄位在看板上
-    def create_Custom_Field(self, params, payload):
+    def post_create_Custom_Field(self, params, payload):
```
python變數codeing style method命名是小寫加下底線post_create_custom_field

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -178,15 +179,15 @@ def create_trello_Custom_Field_on_board(self, board_id):
             "idModel": board_id,
             "modelType": "board",
-            "name": "Priority",
+            "name": trello_Custom_Field_name,
```
python變數codeing style method命名是小寫加下底線post_create_custom_field

VincentSingGo commented on test_trello_api_framework.py:
```diff
@@ -237,34 +238,36 @@ def get_trello_board(self):
-        return board_id
+
+    # 檢查並刪除看板
+    def check_boardID_and_delete_board(self):
```
為啥又有大寫啊python只有class有大寫或env
